service/controller/resource/workermigration/create.go

```diff
@@ -0,0 +1,496 @@
+
+	"github.com/giantswarm/azure-operator/v4/service/controller/key"
+	"github.com/giantswarm/azure-operator/v4/service/controller/resource/workermigration/internal/azure"
+)
+
+// EnsureCreated ensures that built-in workers are migrated to node pool.
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	credentialSecret := &providerv1alpha1.CredentialSecret{
+		Name:      key.CredentialName(cr),
+		Namespace: key.CredentialNamespace(cr),
+	}
+	azureAPI := r.wrapAzureAPI(r.clientFactory, credentialSecret)
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring that built-in workers are migrated to node pool")
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "finding built-in workers VMSS")
+	var builtinVMSS azure.VMSS
+	{
+		builtinVMSS, err = azureAPI.GetVMSS(ctx, key.ResourceGroupName(cr), key.WorkerVMSSName(cr))
+		if azure.IsNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find built-in workers VMSS")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+
+			return nil
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if builtinVMSS.Sku == nil {
+			return microerror.Maskf(executionFailedError, "built-in VMSS Sku is nil")
+		}
+		if builtinVMSS.Sku.Name == nil {
+			return microerror.Maskf(executionFailedError, "built-in VMSS Sku.Name is nil")
+		}
+		if builtinVMSS.Sku.Capacity == nil {
+			return microerror.Maskf(executionFailedError, "built-in VMSS Sku.Capacity is nil")
+		}
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "found built-in workers VMSS")
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring AzureMachinePool CR exists for built-in workers VMSS")
+	azureMachinePool, err := r.ensureAzureMachinePoolExists(ctx, cr, *builtinVMSS.Sku.Name)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensured AzureMachinePool CR exists for built-in workers VMSS")
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring MachinePool CR exists for built-in workers VMSS")
+	machinePool, err := r.ensureMachinePoolExists(ctx, cr, azureMachinePool, int(*builtinVMSS.Sku.Capacity))
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensured MachinePool CR exists for built-in workers VMSS")
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring Spark CR exists for built-in workers VMSS")
+	_, err = r.ensureSparkExists(ctx, cr)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensured Spark CR exists for built-in workers VMSS")
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "finding if node pool workers are ready")
+	if !machinePool.Status.InfrastructureReady || (machinePool.Status.Replicas != machinePool.Status.ReadyReplicas) {
```

Comment:
`machinePool.Status.InfrastructureReady` is set by some handler, copying whatever is in `azureMachinePool.Status.Ready`. Why not using `azureMachinePool.Status.Ready` directly instead?

Reply:
No, this is fine.